### ISSUE TITLE
centos 7  import openssl11 

### ISSUE THREAD

**Issue by sg4r** [sg4r/proxmox-backup-client#16]
on centos 7 by default openssl version is 1.0
To build version 1.0.11 needs openssl version 1.1.0 or higher for openssl :: pkcs5 :: scrypt function
how to configure this?

DEP_OPENSSL_VERSION_NUMBER=0x1_01_01_00_0
does not work

build/main.rs have this config
enum Version {
    Openssl11x,
    Openssl10x,
    Libressl,
}

**Comment by sg4r** [comment]
[dependencies]
openssl = { version = "0.10", features = ["vendored"] }

from https://docs.rs/openssl/0.10.33/openssl/
read making a *-sys crate https://kornel.ski/rust-sys-crate

**Comment by wmorgue** [comment]
Hi @sg4r 👋🏻

My comrade @skazochnik97 and I solved the issue with the certificate on CentOS 7.

After installing, I got an error below:

```bash
proxmox-backup-client version --repository our_proxmox.example:8007

could not connect to server - error trying to connect: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1915:
client version: 1.0.11
```

Hm 🤔 Let's start debugging via `strace`:

```bash
strace proxmox-backup-client version --repository our_proxmox.example:8007
# long long stdout and something interesting

open("/home/centos/proxmox-backup-client/proxmox-backup/target/release/build/openssl-sys-534eff37ea26a8ef/out/openssl-build/install/ssl/openssl.cnf", O_RDONLY) = -1 ENOENT (No such file or directory)
```

> No such file or directory

The Rust language target build directory with `openssl.cnf` config file.

---

### 2-step solution

Create the missing directory:

```bash
mkdir -p /home/centos/proxmox-backup-client/proxmox-backup/target/release/build/openssl-sys-534eff37ea26a8ef/out/openssl-build/install/ssl
```

And finally the last magical step:

```bash
ln -s /etc/ssl/certs/ca-bundle.crt /home/centos/proxmox-backup-client/proxmox-backup/target/release/build/openssl-sys-534eff37ea26a8ef/out/openssl-build/install/ssl/cert.pem
```

Let's try again and we'll see:

```bash
proxmox-backup-client version --repository our_proxmox.example:8007
Password for "root@pam": *****************************************
client version: 1.0.11
server version: 1.1.1
```

That's all 🎉 
Maybe the problem is in the build of the rpm package.


Environment: CenOS 7, OpenSSL 1.0.2k-fips  26 Jan 2017 and `proxmox-backup-1.0.11-2.x86_64.el7.rpm`.

**Comment by charliehu** [comment]
Im also can reproducate this issue.

**Comment by francescor** [comment]
wmorgue what a genius!!!  it works!

**Comment by francescor** [comment]
Please note that the compiled version has this issue for VM with only one CPU https://forum.proxmox.com/threads/using-backup-client-from-within-1-core-vm-stuck-with-futex_wait_private.86136/

solved with the v1.0.6-beta release  (client version: 1.0.6 )

**Comment by sg4r** [comment]
hi, i find where set variable to use openssl11 from epel for build proxmox-backup-client
yum install openssl11 openssl11-libs

export OPENSSL_INCLUDE_DIR=/usr/include/openssl11/
export OPENSSL_LIB_DIR=/usr/lib64/openssl11/
